### PR TITLE
Preparing wxDVC on OSX for resizing implementation

### PR DESCRIPTION
**Pull request by oneeyeman1**

### CODE REVIEW

**Comment by oneeyeman1** [comment]
It would be nice to have a review from Stefan.
Only Cocoa is affected.

**Review by csomor — COMMENTED** [review]
Apart from that small note. I don't see anything obviously wrong. But I'm not really fluent in DataView. So I cannot judge whether this is the right way. What is the problem, this PR solves ?

**Comment by oneeyeman1** [comment]
Stefan,
I'm trying to see if I can do that: https://groups.google.com/forum/?fromgroups#!topic/wx-dev/8xYGk4mUsSs so that OSX behaves like MSW and GTK.

If you have a better idea or maybe you already have a code that I can try I will be happy to test and/or implement.

Thank you.

File: include/wx/dvrenderers.h [hunk 1]
```diff
@@ -317,9 +317,6 @@ class WXDLLIMPEXP_CORE wxDataViewCustomRendererBase
     // Render the item using the current value (returned by GetValue()).
     virtual bool Render(wxRect cell, wxDC *dc, int state) = 0;
 
-    // Return the size of the item appropriate to its current value.
-    virtual wxSize GetSize() const = 0;
-
```
Comment (vadz) [inline include/wx/dvrenderers.h]: Why did you remove this? This (expectedly) broke the builds everywhere but macOS, yet you somehow don't seem at all to be bothered by this... I just don't understand your thinking here.

File: src/osx/cocoa/dataview.mm [hunk 2]
```diff
@@ -3100,6 +3129,12 @@ void UpdateWithRow(int row)
     return true;
 }
 
+wxSize wxDataViewChoiceRenderer::GetSize() const
+{
+    wxSize size(0, 0);
```
Comment (vadz) [inline src/osx/cocoa/dataview.mm]: This is quite obviously wrong, why do this?

**Review by vadz — CHANGES_REQUESTED** [review]
This can't be applied because it breaks the build, of course, but beyond this, I have no idea what is this PR trying to do. You've added Mac-specific `GetSize()` which is not used anywhere, why?

If you do plan to use it in Mac-specific code, you should:

1. Rename it to `OSXGetSize()` or something like this
2. Actually add code using it to show how it's going to be used.

(2) can be done as a separate commit, of course, but it should be part of the same PR. Right now it doesn't make sense to merge it at all, even if it didn't break the build.

**Comment by oneeyeman1** [comment]
Vadim,
> This can't be applied because it breaks the build, of course, but beyond this, I have no idea what is this PR trying to do. You've added Mac-specific `GetSize()` which is not used anywhere, why?

I thought of 2 PRs: add the GetSize() to all renderers and second one to use them.

> 
> If you do plan to use it in Mac-specific code, you should:
> 
>     1. Rename it to `OSXGetSize()` or something like this

No I don't. Because it is done so in MSW code without MSWGetSize() in all generic wxDVC renderer.

> 
>     2. Actually add code using it to show how it's going to be used.
> 
> 
> (2) can be done as a separate commit, of course, but it should be part of the same PR. Right now it doesn't make sense to merge it at all, even if it didn't break the build.

I will put everything in 1 PR making 2 commits.

I guess here its easier to manage with 1 feature PR that is broken by different commits...

Thank you.

**Comment by vadz** [comment]
> > This can't be applied because it breaks the build, of course, but beyond this, I have no idea what is this PR trying to do. You've added Mac-specific `GetSize()` which is not used anywhere, why?
> 
> I thought of 2 PRs: add the GetSize() to all renderers and second one to use them.

This wouldn't be great because we would end up with a bunch of unused functions in the meanwhile. But it still doesn't explain why did you _remove_ `GetSize()` in this PR, breaking the builds.
 
> > If you do plan to use it in Mac-specific code, you should:
> > 1. Rename it to `OSXGetSize()` or something like this
> 
> No I don't. Because it is done so in MSW code without MSWGetSize() in all generic wxDVC renderer.

There is a difference: `GetSize()` is a method of `wxDataViewCustomRendererBase` under MSW, i.e. it only exists in the custom renderers. Under macOS you need it for all renderers, including the native ones, so this is not quite the same method.

**Pull request closed without merging**